File: Source/Compiletime/Composestar/Repository/LanguageModel/ConditionExpressions/And.java
package Composestar.Repository.LanguageModel.ConditionExpressions;

import Composestar.Repository.LanguageModel.ConditionExpressions.Visitor.*;     

/**
 * Summary description for And.
 */
public class And extends ConditionExpression implements IVisitable 
{
	
	public And()
	{
		//
		// TODO: Add Constructor Logic here
		//
	}

	private int _branchId;

	/** @property */
	public int get_BranchId()
	{
		return _branchId;
	}

	/** @property */
	public void set_BranchId(int value)
	{
		_branchId = value;
	} 
	
	private ConditionExpression _left;

	/** @property */
	public ConditionExpression get_Left()
	{
		return _left;
	}

	/** @property */
	public void set_Left(ConditionExpression value)
	{
		_left = value;
	}

	private ConditionExpression _right;

	/** @property */
	public ConditionExpression get_Right()
	{
		return _right;
	}

	/** @property */
	public void set_Right(ConditionExpression value)
	{
		_right = value;
	}

	public void Accept(IVisitor visitor)
	{		
		((IVisitable)_left).Accept(visitor);
		visitor.VisitAndLeft(this);
		((IVisitable)_right).Accept(visitor);
		visitor.VisitAndRight(this);
	}
    
    
    public String toString(){
        return "(" + _left.toString() + " && " + _right.toString() + ")";
    }
}

File: StarLight/Source/StarLightEntities/WeaveSpec/ConditionExpressions/And.cs
#region Using directives
using System;
using System.Collections.Generic;
using System.Text;
using System.Xml;
using System.Xml.Serialization;
using Composestar.StarLight.Entities.WeaveSpec.ConditionExpressions.Visitor;
#endregion

namespace Composestar.StarLight.Entities.WeaveSpec.ConditionExpressions
{
    /// <summary>
    /// An AND operator operating on two parts.
    /// </summary>
    [Serializable]
    [XmlRoot("And", Namespace = "Entities.TYM.DotNET.Composestar")]
    public class AndCondition : ConditionExpression, IVisitable
    {

        /// <summary>
        /// Initializes a new instance of the <see cref="T:And"/> class.
        /// </summary>
        public AndCondition()
        {

        } // And()

        private int _branchId;

        /// <summary>
        /// Gets or sets the branch id.
        /// </summary>
        /// <value>The branch id.</value>
        [XmlAttribute]
        public int BranchId
        {
            get { return _branchId; } // get
            set { _branchId = value; } // set
        } // BranchId


        /// <summary>
        /// Gets or sets the left.
        /// </summary>
        /// <value>The left.</value>
        public ConditionExpression Left
        {
            get
            {
                return _left;
            } // get
            set
            {
                _left = value;
            } // set
        } // Left

        private ConditionExpression _left;

        /// <summary>
        /// Gets or sets the right.
        /// </summary>
        /// <value>The right.</value>
        public ConditionExpression Right
        {
            get
            {
                return _right;
            } // get
            set
            {
                _right = value;
            } // set
        } // Right

        private ConditionExpression _right;

        /// <summary>
        /// Accepts the specified visitor.
        /// </summary>
        /// <param name="visitor">The visitor.</param>
        public void Accept(IVisitor visitor)
        {
            if (visitor == null)
                throw new ArgumentNullException("visitor"); 

            ((IVisitable)_left).Accept(visitor);
            visitor.VisitAndLeft(this);
            ((IVisitable)_right).Accept(visitor);
            visitor.VisitAndRight(this);
        } // Accept(visitor)

    } // class And
}
